server/src/core/python_odoo_builder.rs

```diff
@@ -409,16 +409,16 @@ impl PythonOdooBuilder {
             // });
             return false;
         }
-        if Rc::ptr_eq(symbol, &base_model_syms[0]) ||
-            Rc::ptr_eq(symbol, &model_syms[0]) ||
-            Rc::ptr_eq(symbol, &transient_syms[0])
+        if base_model_syms.first().map_or(false, |first| Rc::ptr_eq(symbol, first))
+            || model_syms.first().map_or(false, |first| Rc::ptr_eq(symbol, first))
+            || transient_syms.first().map_or(false, |first| Rc::ptr_eq(symbol, first))
         {
             return false;
         }
         if compare_semver(session.sync_odoo.full_version.as_str(), "19.1") >= Ordering::Equal{
             let cached_model_tree = (vec![Sy!("odoo"), Sy!("orm"), Sy!("models_cached")], vec![Sy!("CachedModel")]);
             let cached_model = session.sync_odoo.get_symbol(session.sync_odoo.config.odoo_path.as_ref().unwrap(), &cached_model_tree, u32::MAX);
-            if cached_model.is_empty() || Rc::ptr_eq(symbol, &cached_model[0]){
+            if cached_model.first().map_or(false, |first| Rc::ptr_eq(symbol, first)){
```

Comment:
Perso, I find this hard to read (although it's clever).
```suggestion
           // probably rename `cached_model` a few lines above to `chached_model_syms`, to be coherent with the other ones (base_model_syms, ...), and to avoid the shadowing below.
            let if Some(chached_model) = chached_model.first() {
              if Rc::ptr_eq(symbol, chached_model)) {
                return false;
              }
            }
```

If you like this suggestion, consider changing the similar pattern above.

Reply:
I think the suggestion is good here, but it does not work well if we want to do the same check 2 times as we have above. 

What I suggest I do, slightly more readable is to use `cached_model.first().is_some_and(|first| Rc::ptr_eq(symbol, first))`  hoping that `is_some_and` is better than `map_or(false,...)`